pkg/volume/util/operationexecutor/operation_generator.go

```diff
@@ -632,13 +630,31 @@ func (og *operationGenerator) resizeFileSystem(volumeToMount VolumeToMount, devi
 				return nil, nil
 			}
 
-			diskFormatter := &mount.SafeFormatAndMount{
-				Interface: mounter,
-				Exec:      og.volumePluginMgr.Host.GetExec(expandableVolumePlugin.GetPluginName()),
-			}
+			var resizeStatus bool
+			var resizeErr error
 
-			resizer := resizefs.NewResizeFs(diskFormatter)
-			resizeStatus, resizeErr := resizer.Resize(devicePath, deviceMountPath)
+			fsResizablePlugin, callExpandFS := expandableVolumePlugin.(volume.FSResizableVolumePlugin)
+			if callExpandFS {
+				// if any plugin implements FSResizableVolumePlugin interface then call its ExpandFS method
+				// before that if the volume is "kubelet expandable" - like a flex volume, then first call
+				// its ExpandVolumeDevice method before procceeding to ExpandFS
+				if volumeToMount.VolumeSpec.IsKubeletExpandable() {
+					_, resizeErr = fsResizablePlugin.ExpandVolumeDevice(volumeToMount.VolumeSpec, pvSpecCap, pvcStatusCap)
+				}
+				if resizeErr == nil {
+					// kubelet expandable plugins like flex volume may indicate absence of ExpandFS implementation
+					// by returning GenericResizeFilesystemError - so it will be handled by generic resize block below
+					resizeErr = fsResizablePlugin.ExpandFS(volumeToMount.VolumeSpec, devicePath, deviceMountPath)
+				}
+				if resizeErr == nil {
+					resizeStatus = true
+				}
+			}
+			if !callExpandFS || util.IsGenericResizeRequiredError(resizeErr) {
```

Comment:
hrm, flexvolume plugin implementation details should not leak into operation_generator.go . I am sure if plugin did not provide an `ExpandFS` implementation, the default implementation can be called from flexvolume plugin code itself rather than branching out here.

Reply:
@gnufied OK fair enough. I will make sure the defaults implementation of flexvolume as well as others, calls generic resize directly.